### PR TITLE
Element-R: Add current version of the rust-sdk and vodozemac

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

Related to https://github.com/vector-im/element-web/issues/26386

---
Here's what your changelog entry will look like:

## ✨ Features
 * Element-R: Add current version of the rust-sdk and vodozemac ([\#3825](https://github.com/matrix-org/matrix-js-sdk/pull/3825)). Contributed by @florianduros.

### CODE REVIEW

File: src/crypto/index.ts [hunk 1]
```diff
@@ -610,6 +610,15 @@ export class Crypto extends TypedEventEmitter<CryptoEvent, CryptoEventHandlerMap
         this.backupManager.checkAndStart();
     }
 
+    /**
+     * Implementation of {@link CryptoApi#getVersion}.
+     */
+    public getVersion(): null {
+        // `Crypto.getOlmVersion()` is used for the old crypto
```
Comment (florianduros) [inline src/crypto/index.ts:617]: The old crypto version is set in this way https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/client.ts#L2281 and we are using it here https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/components/views/settings/tabs/user/HelpUserSettingsTab.tsx#L75

I keep for the old crypto, the `olmVersion` field and I'm adding a new way to get the version for the new crypto (used here https://github.com/matrix-org/matrix-react-sdk/pull/11785/files#diff-c612bdd14b2fc22382b3675abd80d0b85f887453389d2a4c5b0fdcf2df887bd2R88).

**Review by florianduros — COMMENTED** [review]
I'll ask for an exception about the code coverage. I can add a dummy test to see if the old crypto returns `null` however imo, there is no real gain.

File: src/crypto/index.ts (hunk 1 above)
Comment (richvdh) [reply src/crypto/index.ts:617]: could you make this return something like `Olm ${olmVersionTuple[0]}.${olmVersionTuple[1]}.${olmVersionTuple[2]}`, so that both `Crypto` implementations have a `.getVersion` that returns a string, and just use that in the rust SDK?

Comment (florianduros) [reply src/crypto/index.ts]: Yes good idea!

File: src/crypto-api.ts [hunk 2]
```diff
@@ -39,6 +39,15 @@ export interface CryptoApi {
      */
     globalBlacklistUnverifiedDevices: boolean;
 
+    /**
+     * Return the current version of the crypto module.
+     * Rust crypto: the format is `Rust SDK ${versions.matrix_sdk_crypto}, Vodozemac ${versions.vodozemac}`
+     * Old Crypto: the format is `Olm x.x.x`
+     * XXXX: remove old crypto when we remove support for it
```
Comment (richvdh) [inline src/crypto-api.ts:46]: Nit: this isn't terribly legible as it stands (see https://pr3825--js-sdk-docs-previews.netlify.app/stable/interfaces/crypto.cryptoapi#getVersion): it could do with some paragraph breaks.

Anyway, I feel like we should just give an example, rather than documenting this in a way that makes it sound like someone can rely on that exact format.

```suggestion
     * Return the current version of the crypto module.
     *
     * For example: `Rust SDK ${versions.matrix_sdk_crypto}, Vodozemac ${versions.vodozemac}`.
     *
```

File: spec/unit/crypto.spec.ts [hunk 3]
```diff
@@ -115,6 +115,16 @@ describe("Crypto", function () {
         expect(Crypto.getOlmVersion()[0]).toEqual(3);
     });
 
+    it("should return the current version of the olm library", async () => {
```
Comment (richvdh) [inline spec/unit/crypto.spec.ts:118]:
```suggestion
    it("getVersion() should return the current version of the olm library", async () => {
```

File: spec/unit/rust-crypto/rust-crypto.spec.ts [hunk 4]
```diff
@@ -143,6 +143,13 @@ describe("RustCrypto", () => {
             10000,
         );
 
+        it("should return the current version of the rust sdk and vodozemac", async () => {
```
Comment (richvdh) [inline spec/unit/rust-crypto/rust-crypto.spec.ts:146]:
```suggestion
        it("getVersion() should return the current version of the rust sdk and vodozemac", async () => {
```

**Review by richvdh — APPROVED** [review]
lgtm otherwise
